content/realtime/channels-messages.textile

```diff
@@ -376,7 +388,7 @@ h3(#handling-failures). Handling channel failures
 
 Channel attach and detach operations are synchronous - after initiating an attach request, the client must wait for a response from Ably that confirms that the channel is established on the service.
 
-There are cases where an attach will fail deterministically, such as if the client doesn't have the necessary rights to access the channel.
+The client libraries will attempt to automatically recover from non-fatal error conditions. However, you can handle them yourself if you prefer. <span lang="jsall">For explicit attaches with a callback, both fatal and nonfatal errors will result in the callback being called with an @ErrorInfo@ argument</span>
```

Comment:
Not sure what you're thinking specifically? I mean, once other languages do 0.9 they can replace the js-specific bits with their own language equivalents (e.g. `the callback being called` with `<span lang="jsall">the callback being called</span><span lang="ruby">the onError block being yielded</span>` as necessary, but here I was focusing just on js